hy/core/language.hy

```diff
@@ -369,12 +369,15 @@
   "Chunks coll into n-tuples (pairs by default). The remainder, if any, is not
    included unless a fillvalue is specified. The step defaults to n, but can be
    more to skip elements, or less for a sliding window with overlap."
-  (setv
-   step (or step n)
-   slices (genexpr (itertools.islice coll start None step) [start (range n)]))
-  (if (is fillvalue -sentinel)
-    (apply zip slices)
-    (apply zip-longest slices {"fillvalue" fillvalue})))
+  (if (< n 1) [] (do
```

Comment:
A list of what length though? Even if you attempt to eat a finite iterator by taking zero-sized bites, you'll never finish the meal. On the other hand, it's lazy. This actually does work in Clojure:
```Clojure
user=> (take 10 (partition 0 [1 2 3]))
(() () () () () () () () () ())
```